README.md

```diff
@@ -1,4 +1,750 @@
+		"github.com/IBM/go-sdk-core/v4/core"
+	)
+	
+	// End Code Setup
+
+
+
+### Authentication
+---
+Use one of the following methods to authenticate:
+
+- To authenticate using Basic Auth:
+  Place these values into the Authorization header of the HTTP request in the form Basic <credentials> 
+  where <credentials> is the username and password joined by a single colon `:` base64 encoded. 
+  For example:
+  ```sh
+  echo -n "token:<APIKEY>" | base64
+  ```
+
+- To authenticate using a bearer token:
+  To obtain your token using the IBM Cloud CLI, first log in to IBM Cloud, then run the following command:
+  ```
+  ibmcloud iam oauth-tokens
+  ```
+  Place this token in the Authorization header of the HTTP request in the form Bearer. Both API key or JWT tokens are supported.
+
+- To authenticate directly using the api_key:
+  Place the key directly as the value of the X-Auth-Token HTTP header.
+
+#### Example
+
+Here's an example of how to create the authenticator using either an API key or a BEARER_TOKEN
+
+		// Create Authenticator
+		var authenticator core.Authenticator
+	
+		if apiKey != "" {
+			var err error
+			// Create an Basic IAM authenticator.
+			authenticator, err = core.NewBasicAuthenticator("token", apiKey)
+			if err != nil {
+				fmt.Printf("failed to create new basic authenticator: %s\n", err.Error())
+				os.Exit(1)
+			}
+		} else {
+			var err error
+			// Create an IAM Bearer Token authenticator.
+			authenticator, err = core.NewBearerTokenAuthenticator(bearerToken)
+			if err != nil {
+				fmt.Printf("failed to create new bearer token authenticator: %s\n", err.Error())
+				os.Exit(1)
+			}
+		}
+		// End Authenticator
+
+
+
+
+### Creating a client for the Admin REST API.
+---
+Create a new service object.
+
+		// Create Service
+		serviceAPI, serviceErr := adminrestv1.NewAdminrestV1(&adminrestv1.AdminrestV1Options{
+			URL:           URL,
+			Authenticator: authenticator,
+		})
+		// End Create Service
+
+
+
+### Creating a Kafka topic
+---
+To create a Kafka topic the admin REST SDK issues a POST request to the /admin/topics path. 
+The body of the request contains a JSON document, for example:
+{
+    "name": "topicname",
+    "partitions": 1,
+    "configs": {
+        "retentionMs": 86400000,
+        "cleanupPolicy": "delete"
+    }
+}
+
+The only required field is name is `Partitions` which is defaults to 1 if not set.
+
+Expected HTTP status codes:
+
+- 202: Topic creation request was accepted.
+- 400: Invalid request JSON.
+- 403: Not authorized to create topic.
+- 422: Semantically invalid request.
+
+If the request to create a Kafka topic succeeds then HTTP status code 202 (Accepted) is returned. If the operation fails then a HTTP status code of 422 (Un-processable Entity) is returned, and a JSON object containing additional information about the failure is returned as the body of the response.
+
+
+
+
+#### Example
+
+	func createTopic(serviceAPI *adminrestv1.AdminrestV1) error {
```

Comment:
We should probably wrap all these code examples in code blocks

Reply:
added code blocks